python/GafferUI/NumericPlugValueWidget.py

```diff
@@ -89,17 +89,23 @@ def _updateFromValues( self, values, exception ) :
 		value = sole( values )
 		with Gaffer.Signals.BlockedConnection( self.__valueChangedConnection ) :
 
+			# Animated plugs can only hold static values, which are fast to compute. There's
+			# no need to use placeholder values in that case.
+			hasStaticValue = False
+			if self.getPlugs() and all( Gaffer.Animation.isAnimated( p ) for p in self.getPlugs() ) and len( values ) == 0 :
+				hasStaticValue = True
+
 			# Always give the widget a value, even if we have multiple, because
 			# the _type_ (int or float) is important, and affects interaction
 			# with the widget.
 			if value is not None :
 				self.__numericWidget.setValue( value )
-			elif self.getPlugs() :
+			elif self.getPlugs() and not hasStaticValue :
 				self.__numericWidget.setValue( next( iter( self.getPlugs() ) ).defaultValue() )
 
 			# But if there are multiple values or an error, clear the actual
 			# display so we don't show anything misleading.
-			if exception is not None or value is None :
+			if ( exception is not None or value is None ) and not hasStaticValue :
```

Comment:
I think this could be clearer. My original logic in here was already pretty hard to follow, and mixing extra conditions into the existing stuff doesn't help. `hasStaticValue` is also a misleading name - non-animated plugs have static values, and animated plugs have dynamic values that just happen to be fast to compute.

I think we can deal with this as a single early-out at the very start of the function before we deal with anything else :

```
if len( values ) == 0 and exception is None and self.getPlugs() :
    # Placeholder update for a pending background compute. If we're animated,
    # we don't want to use a `---` placeholder because it messes up editing for
    # the user. Animation is quick enough to compute that there is no need
    # for a placeholder anyway, so we'll just wait for the final update.
    if all( Gaffer.Animation.isAnimated( p ) for p in self.getPlugs() ) :
        return
```

Does that work?